File: 205-isomorphic-strings/205-isomorphic-strings.go
func isIsomorphic(s string, t string) bool {
    st := make(map[byte]byte)
    ts := make(map[byte]byte)
    
    if len(s) != len(t) { return false}
     
    for i := 0; i < len(s); i ++  {
        _, isInST := st[s[i]]
        _, isInTS := ts[t[i]]
        
        if (!isInST && !isInTS) {
            st[s[i]] = t[i]
            ts[t[i]] = s[i]
            continue
        }
        
        if isInST && st[s[i]]!=t[i] { return false }
        if isInTS && ts[t[i]]!=s[i] { return false}
    }
    
    return true
}

File: 205-isomorphic-strings/205-isomorphic-strings.ts
function isIsomorphic(s: string, t: string): boolean {
    
    const st = new Map()
    const ts = new Map()
    
    let i
    
    for ( i = 0; i < s.length; i++ ){
        
        if (!st.has(s[i]) && !ts.has(t[i])) {
            st.set(s[i], t[i])
            ts.set(t[i], s[i])
            continue
        }
        
        if (st.has(s[i]) && st.get(s[i]) !== t[i]) {break}
        
        if (ts.has(t[i]) && ts.get(t[i]) !== s[i]) {break}
    }
    
    //console.log({f: f.get(s[i]), t: t[i], i, fe: f.entries()})
    
    return (st.get(s[i]) === t[i]) && (ts.get(t[i]) === s[i])
};